File: leetcode/21/MergeList.java
class ListNode {
    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if (l1 == null)
            return l2;
        if (l2 == null)
            return l1;
        ListNode head = null, tail = null;
        if (l1.val <= l2.val) {
            head = tail = l1;
            l1 = l1.next;
        } else {
            head = tail = l2;
            l2 = l2.next;
        }
        while (l1 != null && l2 != null) {
            if (l1.val <= l2.val) {
                tail.next = l1;
                tail = l1;
                l1 = l1.next;
            } else {
                tail.next = l2;
                tail = l2;
                l2 = l2.next;
            }
        }
        if (l1 == null) {
            tail.next = l2;
        } else
            tail.next = l1;
        return head;
    }
}

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution1 {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        
        ListNode dummyHead = new ListNode(0);
        ListNode current = dummyHead;
        
        while((l1 != null) || (l2 != null) ){
            
            if(l1 == null)
            {
                current.next = new ListNode(l2.val);
                current = current.next;
                l2 = l2.next;
            }
            else if(l2 == null)   
            {
                current.next = new ListNode(l1.val);
                current = current.next;
                l1 = l1.next;
            }else{
                if(l1.val < l2.val)
                {
                    current.next = new ListNode(l1.val);
                    current = current.next;
                    l1 = l1.next;
                }else
                {
                    current.next = new ListNode(l2.val);
                    current = current.next;
                    l2 = l2.next;
                }      
            }    
        }
        return dummyHead.next;
    }
}

File: leetcode/21/MergeList.js
const mergeTwoLists = (l1, l2) => {
    let dummy = new ListNode(-1)
    let head = dummy
    while(l1 !== null && l2 !== null){
        if(l1.val < l2.val){
            dummy.next = l1
            l1 = l1.next
        } else {
            dummy.next = l2
            l2  = l2.next
        }
        dummy = dummy.next
    }
      
      if(l1 !== null){
          dummy.next = l1 
      }
      if(l2 !== null){
          dummy.next = l2
      }
      return head.next
  };


  /**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists1 = function(l1, l2) {
    let dummyHead = new ListNode(0);
    let current = dummyHead;
    
    while(l1 != null && l2 !=null)
    {
        if(l1.val < l2.val)
        {
            current.next = l1;
            l1 = l1.next;
        }else
        {
            current.next = l2;
            l2 = l2.next;
        }
        current = current.next;
    }
    
    if(l1 != null)
    {
        current.next = l1;
        l1 = l1.next;
    }
    
    
    if(l2 != null)
    {
        current.next = l2;
        l2 = l2.next;
    }
    
    return dummyHead.next;
};